konrad/core.py

```diff
@@ -55,33 +55,36 @@ def __init__(
         diurnal_cycle=False,
         co2_adjustment_timescale=np.nan,
         logevery=None,
+        timestep_adjuster=None,
     ):
         """Set-up a radiative-convective model.
 
         Parameters:
 
             atmosphere: :py:class:`konrad.atmosphere.Atmosphere`.
 
-            timestep (float or str): Model time step (per iteration)
+            timestep (float, str or timedelta): Model time step
 
                 * If float, time step in days.
                 * If str, a timedelta string (see :func:`konrad.utils.parse_fraction_of_day`).
+                * A `timedelta` object is directly used as timestep.
 
-            max_duration (float or str): Maximum duration of the simulation
+            max_duration (float, str or timedelta): Maximum duration.
                 The duration is given in model time
 
                 * If float, maximum duration in days.
                 * If str, a timedelta string (see :func:`konrad.utils.parse_fraction_of_day`).
+                * A `timedelta` object is directly used as timestep.
 
             outfile (str): netCDF4 file to store output.
 
             experiment (str): Experiment description (stored in netCDF output).
 
-            writeevery (int, float or str): Set output frequency.
+            writeevery (float, str or timedelta): Set output frequency.
 
-                * int: Every nth iteration
                 * float: Every nth day in model time
```

Comment:
I think it would be nice to have the option to output at every iteration, especially if the timestep is changing, as this would allow the user to a) see how the timestep changes and b) capture the behaviour better, eg by having higher frequency output for the stratospheric adjustment and lower frequency output when it is close to equilibrium and the temperatures are not changing much.

Reply:
Another nice option may be to store the timestep size whenever the output is written out.

Reply:
Setting the timestep to `datetime.timedelta(0)` (or `"0h"`) should do the trick. I will test that and update the docstring accordingly. I guess that is easier than implementing a special case like `-1` or keeping track of the iteration using integers